### PR TITLE
Adding adKernel adapter

### PR DESCRIPTION
**Pull request by ckbo3hrk**
## Type of change

- [x] New bidder adapter
## Description of change

Added AdKernel  adapter

- test parameters for validating bids

```
{
  bidder: 'adkernel',
  params: {
    zoneId : '30164',
    host : 'cpm.metaadserving.com'
  }
}
```
- contact email of the adapter’s maintainer: denis@adkernel.com
- [x] official adapter submission

### CODE REVIEW

File: src/adapters/adkernel.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
+var bidmanager = require('../bidmanager.js');
```
Comment (mkendall07) [inline src/adapters/adkernel.js]: please use ES 6 import syntax here.

File: src/adapters/adkernel.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+var bidfactory = require('../bidfactory.js');
+var ajax = require('../ajax.js');
+var utils = require('../utils.js');
+
+/**
+ * Adapter for requesting bids from AdKernel white-label platform
+ * @class
+ */
+var AdkernelAdapter = function AdkernelAdapter() {
```
Comment (mkendall07) [inline src/adapters/adkernel.js]: prefer `const`

File: src/adapters/adkernel.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+  }
+
+  /**
+   *  Process all bids grouped by network/zone
+   */
+  function processBids(bids) {
+    const dispatcher = new RtbRequestDispatcher();
+    //process individual bids
+    utils._each(bids, function (bid) {
+      let size = bid.sizes[0];
```
Comment (mkendall07) [inline src/adapters/adkernel.js]: Does your adapter support multi size?

File: src/adapters/adkernel.js [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+      'ad_type': 'rtb',
+      'r': encodeURIComponent(JSON.stringify(rtbReq))
+    };
+  }
+
+  /**
+   *  Create bid object for the bid manager
+   */
+  function createBidObject(bidObj, width, height) {
+    return utils.extend(bidfactory.createBid(1), {
```
Comment (mkendall07) [inline src/adapters/adkernel.js]: Please store a reference to the bid request object and pass it to `createBid(1, bidRequest)` as this allows us to match the bid.adId to bid response.

File: test/spec/adapters/adkernel_spec.js [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+    sizes: [[728, 90]]
+  }, bid3_host2 = {
+    bidder: "adkernel",
+    bidId: "Bid_02",
+    params: {zoneId: 1, host: "rtb-private.adkernel.com"},
+    placementCode: "ad-unit-2",
+    sizes: [[728, 90]]
+  };
+
+	const bidResponse1 = {
```
Comment (mkendall07) [inline test/spec/adapters/adkernel_spec.js]: nit: please apply formatting to spec file.

File: src/adapters/adkernel.js [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+   */
+  function RtbRequestDispatcher() {
+    const imp2adUnit = {},
+      _dispatch = {},
+      site = createSite();
+
+    //translate adunit info into rtb impression dispatched by host/zone
+    this.addImp = function (host, zone, width, height, bidId, adUnitId) {
+      if (!(host in _dispatch)) {
+        _dispatch[host] = {};
```
Comment (mkendall07) [inline src/adapters/adkernel.js]: If `host` is undefined, you should abort this request. Without the host param, I was seeing network requests to `http://undefined/....`

File: src/adapters/adkernel.js [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+  /**
+   * Build ad-network specific endpoint url
+   */
+  function buildEndpointUrl(host) {
+    return window.location.protocol + '//' + host + '/rtbg';
+  }
+
+  function buildRequestParams(zone, rtbReq) {
+    return {
+      'zone': zone,
```
Comment (mkendall07) [inline src/adapters/adkernel.js]: `zone` should be encoded too if possible.

**Comment by mkendall07** [comment]
@ckbo3hrk 
Any update here?

**Comment by ckbo3hrk** [comment]
@mkendall07 
On vacation. Everything will be done on 14th of November.

File: src/adapters/adkernel.js (hunk 3 above)
Comment (ckbo3hrk) [inline src/adapters/adkernel.js]: No, adapter doesn't support multiple sizes.

**Comment by ckbo3hrk** [comment]
@mkendall07 
Code review comments has been adressed.

File: src/adapters/adkernel.js [hunk 8]
```diff
@@ -0,0 +1,207 @@
+import bidmanager from "src/bidmanager";
```
Comment (mkendall07) [inline src/adapters/adkernel.js]: please use single quotes for string literals `'`

**Comment by mkendall07** [comment]
@ckbo3hrk if you get this done today, it will be included in the 0.15.0 release.

**Comment by ckbo3hrk** [comment]
@mkendall07 
done

**Comment by ckbo3hrk** [comment]
@mkendall07 
done

**Comment by mkendall07** [comment]
LGTM. Bids are being received, unit tests pass. Thanks for contributing to Prebid.js

**Comment by aneuway2** [comment]
@ckbo3hrk are all parameter's required or are there optional inputs?

**Comment by ckbo3hrk** [comment]
@aneuway2
Both parameters are required.

**Comment by mkendall07** [comment]
@ckbo3hrk 
Please submit a docs PR here: https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders

**Pull request merged**